src/main/groovy/io/seqera/wave/service/k8s/K8sServiceImpl.groovy

```diff
@@ -431,13 +432,29 @@ class K8sServiceImpl implements K8sService {
      */
     @Override
     V1ContainerStateTerminated waitPod(V1Pod pod, long timeout) {
+        return waitPod(pod, pod.metadata.name, timeout)
+    }
+
+    /**
+     * Wait for a pod a completion
+     *
+     * @param pod
+     *      The pod name
+     * @param timeout
+     *      Max wait time in milliseconds
+     * @return
+     *      An instance of {@link V1ContainerStateTerminated} representing the termination state
+     *      or {@code null} if the state cannot be determined or timeout was reached,
+     */
+    @Override
+    V1ContainerStateTerminated waitPod(V1Pod pod, String containerName, long timeout) {
         final name = pod.metadata.name
```

Comment:
this `name` is not used. Also I wonder if it's really needed to pass the `containerName` as argument because the `pod` should already provide that info via `pod.metadata.name`

Reply:
in this case pod name and container name is different, pod name derives from job name and wave specifies the container name

Reply:
i have removed name. this is overloaded function when using job because in case of job we cannot set pod name, pod names get auto generated with jodname-<string>, so we need to pass container name as different argument

Reply:
Yes, but I was thinking if we really need that `find` 

https://github.com/seqeralabs/wave/blob/947e6790b053ee5d5dad8722deba4f0e71733f26/src/main/groovy/io/seqera/wave/service/k8s/K8sServiceImpl.groovy#L457

at the end are always used pod or job with exactly one container, therefore I believe `containerStatuses` will always be one item list. 

Am I wrong?

Reply:
I'm going to simplify a bit

Reply:
Made a new release in stage, give it a try at your convenience

Reply:
checked, job are created and pods too